lib/active_merchant/billing/gateways/paypal_checkout.rb

```diff
@@ -0,0 +1,376 @@
+        requires!({ order_id: order_id }, :order_id)
+
+        post = {}
+        add_payment_source(options[:payment_source], post) unless options[:payment_source].nil?
+        add_application_context(options[:application_context], post) unless options[:application_context].nil?
+        commit(:post, "v2/checkout/orders/#{ order_id }/capture", post, options[:headers])
+      end
+
+      def refund(capture_id, options={})
+        requires!({ capture_id: capture_id }, :capture_id)
+
+        post = {}
+        add_amount(options[:amount], post) unless options[:amount].nil?
+        add_invoice(options[:invoice_id], post) unless options[:invoice_id].nil?
+        add_note(options[:note_to_payer], post) unless options[:note_to_payer].nil?
+        commit(:post, "v2/payments/captures/#{ capture_id }/refund", post, options[:headers])
+      end
+
+      def void(authorization_id, options={})
+        requires!({ authorization_id: authorization_id }, :authorization_id)
+        post = {}
+        commit(:post, "v2/payments/authorizations/#{ authorization_id }/void", post, options[:headers])
+      end
+
+      def do_capture(authorization_id, options={})
+        requires!(options.merge!({ authorization_id: authorization_id  }), :authorization_id)
+
+        post = {}
+        add_amount(options[:amount], post) unless options[:amount].nil?
+        add_invoice(options[:invoice_id], post) unless options[:invoice_id].nil?
+        add_final_capture(options[:final_capture], post) unless options[:final_capture].nil?
+        add_payment_instruction(options[:payment_instruction], post) unless options[:payment_instruction].nil?
+        add_note(options[:note_to_payer], post) unless options[:note_to_payer].nil?
+
+        commit(:post, "v2/payments/authorizations/#{ authorization_id }/capture", post, options[:headers])
+      end
+
+      def get_order_details(order_id, options={})
+        requires!(options.merge(order_id: order_id), :order_id)
+        commit(:get, "v2/checkout/orders/#{ order_id }", nil, options[:headers])
+      end
+
+      def get_authorization_details(authorization_id, options={})
+        requires!(options.merge(authorization_id: authorization_id), :authorization_id)
+        commit(:get, "v2/payments/authorizations/#{ authorization_id }", nil, options[:headers])
+      end
+
+      def get_capture_details(capture_id, options={})
+        requires!(options.merge(capture_id: capture_id), :capture_id)
+        commit(:get, "v2/payments/captures/#{ capture_id }", nil, options[:headers])
+      end
+
+      def get_refund_details(refund_id, options={})
+        requires!(options.merge(refund_id: refund_id), :refund_id)
+        commit(:get, "v2/payments/refunds/#{ refund_id }", nil, options[:headers])
+      end
+
+      private
+
+      def add_purchase_units(options, post)
+        post[:purchase_units] = []
+        options.map do |purchase_unit|
+          post[:purchase_units] << add_purchase_unit(purchase_unit)

```

Comment:
let's call this get_purchase_unit or something since this returns the value instead of updating the post

Reply:
Renamed it to `construct_purchase_unit` 👍🏻